[cfe/ffi]: Removing `.address` invalid expression in leaf call to prevent errors

codesculpture opened the pull request.

codesculpture committed:
Transforming invalid expression to actual invalid expression by replacing the existing .address expression

codesculpture commented:
I actually tried to use `git cl` but i faced several issues (i.e i solved few but finally the below one which i dint solve).
```
saving CL description to D:\Programming\Binaries\depot_tools\.git_cl_description_backup

info: please complete authentication in your browser...
To https://github.com/codesculpture/sdk.git
 * [new reference]           2059beb8af4682405a0ba5348df44b644fd08205 -> refs/for/refs/heads/main%m=Initial_upload
Traceback (most recent call last):
  File "D:\Programming\Binaries\depot_tools\metrics.py", line 302, in print_notice_and_exit
    yield
  File "D:\Programming\Binaries\depot_tools/git_cl.py", line 7092, in <module>
    sys.exit(main(sys.argv[1:]))
             ^^^^^^^^^^^^^^^^^^
  File "D:\Programming\Binaries\depot_tools/git_cl.py", line 7073, in main
    return dispatcher.execute(OptionParser(), argv)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Programming\Binaries\depot_tools\subcommand.py", line 254, in execute
    return command(parser, args[1:])
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\Programming\Binaries\depot_tools/git_cl.py", line 5324, in CMDupload
    UploadAllSquashed(options, orig_args)
  File "D:\Programming\Binaries\depot_tools/git_cl.py", line 5472, in UploadAllSquashed
    cl.PostUploadUpdates(options, new_upload, change_numbers[i])
IndexError: list index out of range
```

mraleph commented:
@codesculpture you need to clone from https://dart.googlesource.com/sdk instead of GitHub if you want to use git-cl / Gerrit

codesculpture committed:
Added test for verifying no more cascading errors.

And removed previously added cascading errors (which are wrong), as now it is being fixed hence removing it.

codesculpture committed:
Fixed error position

codesculpture committed:
AsExpression may come up with InvalidExpression hence handling that

codesculpture committed:
Added test cases
